File: python2/tests/lehmer03.py
import sys
sys.path.insert(1, sys.path[0] + '/..')
from permutation import Permutation

for i in range(2, 13):
    for j in range(1,i):
	p = Permutation.transposition(i,j)
	lehmer = p.lehmer
	remhel = Permutation.fromLehmer(lehmer)
	#if p != remhel:
	print '%s -> %d -> %s' % (p, lehmer, remhel)

File: cpp/tests/lehmer03.cpp
#include <iostream>
#include "Permutation.hpp"
using namespace std;

int main(void) {
 for (int i=2; i<13; i++) {
  for (int j=1; j<i; j++) {
   Permutation p = Permutation::transposition(i,j);
   int lehmer = p.lehmer();
   Permutation remhel = Permutation::fromLehmer(lehmer);
   //if (p != remhel)
    cout << string(p) << " -> " << lehmer << " -> " << string(remhel) << endl;
  }
 }
 return 0;
}
